### PR TITLE
Update dependencies and run e2e tests on PRs

### PR DESCRIPTION
**Pull request by Siegrift**

### CODE REVIEW

**Comment by Siegrift** [comment]
Seems that https://www.githubstatus.com/ is down at the moment so I am not sure the workflow is setup correctly

File: README.md [hunk 1]
```diff
@@ -110,8 +103,19 @@ branch:
 #### Updating the name servers
 
 The primary way to access the DAO dashboard is through the `api3.eth` ENS name, which points directly to the IPFS hash.
-Then, the user can either connect to mainnet on their Metamask and visit `api3.eth/` (the recommended way), or they can visit `https://api3.eth.link/` (this is reported to be down frequently, see [this](https://blog.cloudflare.com/cloudflare-distributed-web-resolver/) for more information).
-We have also forwarded `https://dao.api3.org` to the IPFS hash (using the `dweb.link` gateway), but we do not recommend using this unless necessary.
+Then, the user can either connect to mainnet on their Metamask and visit `api3.eth/` (the recommended way), or they can
+visit `https://api3.eth.link/`.
+
+<!-- markdown-link-check-disable -->
+<!-- The link below exists and works, but the github actions check says it does not" -->
```
Comment (andreogle) [inline README.md:110]: I wonder why these links fail 😕

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -19,9 +19,9 @@ import { ethersProvider, HOME_PAGE } from './common';
 Cypress.Commands.add('increaseTimeAndRelogin', (timeInSeconds: number) => {
   cy.log('increaseTimeAndRelogin');
 
-  cy.wrap(ethersProvider.send('evm_increaseTime', [timeInSeconds])).then(() =>
-    cy.clock(Date.now() + 1000 * timeInSeconds, ['Date'])
-  );
+  cy.wrap(
+    ethersProvider.send('evm_increaseTime', [timeInSeconds]).then(() => ethersProvider.send('evm_mine', []))
```
Comment (andreogle) [inline cypress/support/commands.ts:23]: 👍🏻

File: package.json [hunk 3]
```diff
@@ -4,17 +4,20 @@
...
+    "build": "react-scripts build",
+    "ci:cypress": "yarn start-test \"yarn concurrently 'yarn eth:node' 'yarn ci:start'\" '8545|3000' 'yarn eth:deploy-dao-contracts-on-hardhat && yarn cypress:run'",
+    "ci:start": "FORCE_COLOR=true yarn start | cat",
     "cypress:open": "cypress open",
     "cypress:run": "cypress run",
     "eject": "react-scripts eject",
-    "eth:compile-contracts": "cd hardhat && npx hardhat compile",
+    "eth:deploy-dao-contracts-on-hardhat": "node dev-scripts/deploy-dao-contracts-on-hardhat.js",
     "eth:node": "cd hardhat && npx hardhat node --network hardhat",
+    "eth:prepare-dao-contracts-for-hardhat": "node dev-scripts/prepare-dao-contracts-for-hardhat.js",
```
Comment (andreogle) [inline package.json:15]: I would drop the "on-hardhat" and "for-hardhat" parts of the script names but don't feel strongly here

File: package.json [hunk 4]
```diff
@@ -50,11 +53,11 @@
     ]
   },
   "dependencies": {
-    "@sentry/react": "^6.8.0",
-    "@walletconnect/web3-provider": "^1.4.1",
+    "@sentry/react": "^6.11.0",
+    "@walletconnect/web3-provider": "^1.5.3",
```
Comment (andreogle) [inline package.json:57]: 👍🏻

File: src/page-container.tsx [hunk 5]
```diff
@@ -16,7 +16,8 @@ const PageContainer = (props: Props) => {
   // TODO: Implement a nicely designed error page/component for Sentry.ErrorBoundary fallback
   // See: https://docs.sentry.io/platforms/javascript/guides/react/components/errorboundary/
   return (
-    <Sentry.ErrorBoundary fallback="An error has occurred. Please refresh and try again.">
+    // Sentry.ErrorBoundary requires 'fallback' to be React component so we can't pass a string there
```
Comment (andreogle) [inline src/page-container.tsx:19]: Good catch. I thought this was working

File: package.json [hunk 6]
```diff
@@ -63,49 +66,51 @@
...
+    "lint-staged": "^11.1.2",
     "node-sass": "^6.0.1",
     "npx": "^10.2.2",
     "prettier": "^2.3.2",
     "react-scripts": "4.0.3",
     "replace-in-file": "^6.2.0",
     "sort-package-json": "^1.50.0",
-    "ts-node": "^10.0.0",
-    "typechain": "^5.1.1",
+    "start-server-and-test": "^1.13.1",
```
Comment (andreogle) [inline package.json:111]: Where is this used?

**Review by andreogle — DISMISSED** [review]

File: package.json (hunk 6 above)
Comment (Siegrift) [reply package.json:111]: Allows you to call `yarn start-test` which is what I use to run cypress tests on CI (but I am not very successful in it as of now :D )

File: README.md (hunk 1 above)
Comment (Siegrift) [reply README.md:111]: Not sure, but some fail all the time and some are extremely flaky, so I've ignored those ones.

File: package.json (hunk 3 above)
Comment (Siegrift) [reply package.json:15]: I'd keep it this way, because there is another command which "fixes" the DAO contracts just for typechain.

**Review by andreogle — APPROVED** [review]
Great work!

**Pull request merged**